azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/manager/AppServiceKuduManager.java

```diff
@@ -84,20 +92,29 @@ public static AppServiceKuduManager getClient(@Nonnull WebAppBase webAppBase) {
 
         final KuduService kuduService = RestProxy.create(KuduService.class, httpPipeline,
                 SerializerFactory.createDefaultManagementSerializerAdapter());
-        return new AppServiceKuduManager(host, kuduService);
+        return new AppServiceKuduManager(host, kuduService, appService);
     }
 
-    public Mono<byte[]> getFileContent(final String path) {
-        final Flux<ByteBuffer> byteBufferFlux = this.kuduService.getFileContent(host, path).flatMapMany(StreamResponse::getValue);
-        return FluxUtil.collectBytesInByteBufferStream(byteBufferFlux);
+    public Flux<ByteBuffer> getFileContent(final String path) {
+        return this.kuduService.getFileContent(host, path).flatMapMany(StreamResponse::getValue);
     }
 
     public List<? extends AppServiceFile> getFilesInDirectory(String dir) {
         // this file is generated by kudu itself, should not be visible to user.
-        final List<AppServiceFile> files = this.kuduService.getFilesInDirectory(host, dir).block().getValue().stream()
+        final Mono<Response<List<AppServiceFile>>> filesInDirectory = this.kuduService.getFilesInDirectory(host, dir);
+        return getValueFromResponseMono(filesInDirectory, Collections.emptyList()).stream()
                 .filter(file -> !"text/xml".equals(file.getMime()) || !file.getName().contains("LogFiles-kudu-trace_pending.xml"))
+                .map(file -> file.withApp(appService).withPath(Paths.get(dir, file.getName()).toString()))
                 .collect(Collectors.toList());
-        return files;
+    }
+
+    public AppServiceFile getFileByPath(String path) {
+        final File file = new File(path);
```

Comment:
for get a file, why here we list all the files in the parent folder, and then get the target from a large number of files?

Reply:
If we call `api/vfs/{path}/` to file directly, it will download the file not get the file meta data